### ISSUE TITLE
Actual PID frequency is only 1600 hz

### ISSUE THREAD

**Issue by anshchawla521** [anshchawla521/DIY_FC#12]
the actual measured PID frequency is only 1600 hz

**Comment by anshchawla521** [comment]
I used the below code to test the PID frequency
```C
unsigned long TIMES[8] = {0};
void loop()
{
  prev_time = current_time;
  current_time = micros();
  dt = (current_time - prev_time) / 1000000.0;

  // loopBlink(); // Indicate we are in main loop with short blink every 1.5 seconds

  // Print data at 100hz - Can Select multiple:
  // Note: its Important to not comment out the the controlPrintRate function.
  // Note: Its not recommended to go above/below 100 hz print speed

  // printRadioData();
  // printDesiredState();
  //  printGyroData();
  //   printAccelData();
  //   printMagData();
  // printRollPitchYaw();
  // printPIDoutput();
  // printMotorCommands();
  // printBatteryStatus();
  // printCoreTemp();
  // printArmStatus();
  TIMES[0] = micros();
  printPercentageTimeCpuUsed();
  controlPrintRate(100);
  TIMES[1] = micros();

  getIMUdata(); // takes 143 us // improve this
  TIMES[2] = micros();
  // To check if gps code is working fine
  // getGpsData();
  // printGPSData();

  // getCompassData();
  // printCompassData();

  Madgwick();
  getDesiredState();
  TIMES[3] = micros();

  //
  // logData();
  //
  getCommands(); // Pulls current available radio commands
  handelAuxChannels();
  handelFlightMode(); // PID loops goes inside this
  controlMixer();
  TIMES[4] = micros();
  throttleCut();
  checkFailsafe(); // this function is used instead of failsafe in drehmflight
  scaleCommands();
  TIMES[5] = micros();
  commandMotors(); // took 124 us 
  TIMES[6] = micros();
  // failSafe();    // Prevent failures in event of bad receiver connection, defaults to failsafe values assigned in setup
  getBatteryStatus();
  getCoreTemp(); // this all took 332 us
  TIMES[7] = micros();
  //// Regulate loop rate
  loopRate(2000); // Do not exceed 2000Hz, all filter parameters tuned to 2000Hz by default
}
```

and the loop rate function as below 
```C
void loopRate(int freq)
{
  // DESCRIPTION: Regulate main loop rate to specified frequency in Hz
  /*
     It's good to operate at a constant loop rate for filters to remain stable and whatnot. Interrupt routines running in the
     background cause the loop rate to fluctuate. This function basically just waits at the end of every loop iteration until
     the correct time has passed since the start of the current loop for the desired loop rate in Hz. 2kHz is a good rate to
     be at because the loop nominally will run between 2.8kHz - 4.2kHz. This lets us have a little room to add extra computations
     and remain above 2kHz, without needing to retune all of our filtering parameters.
  */
  float invFreq = 1.0 / freq * 1000000.0;
  unsigned long checker = micros();

  percentage_time_used_by_code = ((checker - current_time) / invFreq) * 100;
  // Sit in loop until appropriate time has passed
  while (invFreq > (checker - current_time))
  {
    Serial.println("Wasting time");
    checker = micros();
  }
  Serial.print("times are ");
  for (int i = 0; i < 7; i++)
  {
    Serial.println(TIMES[i + 1] - TIMES[i]);
  }
}```

The above code output 
```
times are 6
130
14
6
2
124
333
times are 7
136
13
7
2
123
332
times are 6
142
25
7
2
124
332
times are 6
136
13
7
2
124
333
times are 6
148
14
6
3
123
336
```

now the culprit turned out to be the analogRead (Coretemp) which was taking 250us approx .

**Comment by anshchawla521** [comment]
SOLUTION: Give priority to functions like PID and read the battery voltage and other parameters only once in a while which assures us that the PID frequency wont be affected

**Comment by anshchawla521** [comment]
Fixed in commit e8f49c7647c3ba2f6389a6515ffd2333e9bdb182
